### PR TITLE
feat: bump GWAPI v1.1 and support for GRPCRoute v1

### PR DESCRIPTION
**Pull request by tao12345666333**
**What this PR does / why we need it**:

* Bump GWAPI to v1.1
* Add support for GRPCRoute v1

**Which issue this PR fixes**:

fixes: https://github.com/Kong/kubernetes-ingress-controller/issues/5960

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -49,13 +51,13 @@ require (
...
-	k8s.io/api v0.29.4
-	k8s.io/apiextensions-apiserver v0.29.3
-	k8s.io/apimachinery v0.29.4
-	k8s.io/client-go v0.29.4
-	k8s.io/component-base v0.29.4
+	k8s.io/api v0.30.0
+	k8s.io/apiextensions-apiserver v0.30.0
+	k8s.io/apimachinery v0.30.0
+	k8s.io/client-go v0.30.0
+	k8s.io/component-base v0.30.0
```
Comment (pmalek) [inline go.mod:58]: Can we please have all of those k8s related version bumps in a separate PR? 

This makes it easier to track changes, backport changes to other releases (if that's ever necessary) and track failing tests to their respective changes.

Comment (tao12345666333) [reply go.mod:58]: @pmalek  Sure.  Please see https://github.com/Kong/kubernetes-ingress-controller/pull/5935

**Comment by tao12345666333** [comment]
Blocked by https://github.com/Kong/kubernetes-ingress-controller/pull/5935

File: CHANGELOG.md [hunk 2]
```diff
@@ -138,6 +138,9 @@ Adding a new version? You'll need three changes:
 - Added `FallbackConfiguration` feature gate to enable the controller to generate a fallback configuration
   for Kong when it fails to apply the configuration. The feature gate is disabled by default.
   [#5993](https://github.com/Kong/kubernetes-ingress-controller/pull/5993)
+- Add support for Kubernetes Gateway API v1.1:
+  - add a flag `--enable-controller-gwapi-grpcroute` to control whether enable or disable GRPCRoute controller.
+  - add support for `GRPCRoute` v1, which requires users to upgrade the Gateway API's CRD to v1.1.
```
[the remark this answers is not in the document]
Comment (tao12345666333) [reply CHANGELOG.md:145]: https://github.com/Kong/kubernetes-ingress-controller/pull/5918/commits/c7b36a6ed7a3cdf30b2ac62e2827481f35f79183 Added

**Review by randmonkey — APPROVED** [review]

**Pull request merged**